quickwit/quickwit-serve/src/otlp_api/rest_handler.rs

```diff
@@ -126,7 +141,16 @@ pub(crate) fn otlp_default_traces_handler(
         .and(with_arg(BodyFormat::default()))
         .map(into_rest_api_response)
 }
-
+/// Open Telemetry REST/Protobuf traces ingest endpoint.
+#[utoipa::path(
+    post,
+    tag = "Open Telemetry",
+    path = "/{index}/otlp/v1/traces",
```

Comment:
I'm a bit surprised that we use a path param here and not a header (also what was suggested in https://github.com/quickwit-oss/quickwit/issues/5413). It would be more in line with the way it works in GRPC. But of course in GRPC you don't have a path, so the comparison doesn't really hold. I guess both are fine!

Reply:
That api endpoint was already there (but didn't work properly and was absent from openapi definiton). @fmassot, looks like that code was yours, what do you think (index in path vs in header)?

Reply:
Customizing the endpoint or adding headers with the OTEL collectors is easy, so let's support both, but this PR is great for a first pass. @trinity-1686a, please go ahead and merge this.